File: 1) INTRODUCCION_A_POO/Ejercicio-2/JAVA/Bus.java
public class Bus {
    int capacidadMaxima;
    int pasajerosActuales;
    double recaudacion;
    final double COSTO_PASAJE = 1.50;

    public Bus(int capacidadMaxima) {
        this.capacidadMaxima = capacidadMaxima;
        this.pasajerosActuales = 0;
        this.recaudacion = 0.0;
    }

    public void subir(int cantidad) {
        int asientosDisponibles = this.capacidadMaxima - this.pasajerosActuales;
        
        if (cantidad <= asientosDisponibles) {
            this.pasajerosActuales += cantidad;
        } else {
            System.out.println("ADVERTENCIA: Solo subieron " + asientosDisponibles + " pasajeros. Bus lleno.");
            this.pasajerosActuales = this.capacidadMaxima;
        }
    }

    public void cobrarPasaje(int cantidad) {
        this.recaudacion += cantidad * COSTO_PASAJE;
    }

    public void mostrarAsientosDisponibles() {
        System.out.println(this.capacidadMaxima - this.pasajerosActuales);
    }
    
    public double getRecaudacion() {
        return this.recaudacion;
    }
    
    public int getPasajerosActuales() {
        return this.pasajerosActuales;
    }

    public static void main(String[] args) {
        Bus miBus = new Bus(50);

        System.out.println("Capacidad Maxima del bus: " + miBus.capacidadMaxima);
        System.out.println("Pasajeros iniciales: " + miBus.getPasajerosActuales());
        System.out.printf("Recaudacion inicial: $%.2f%n", miBus.getRecaudacion());

        System.out.println("\n---  Suben 10 pasajeros ---");
        miBus.subir(10);
        miBus.cobrarPasaje(10);
        System.out.println("Pasajeros actuales: " + miBus.getPasajerosActuales());
        System.out.printf("Recaudacinn total: $%.2f%n", miBus.getRecaudacion());
        System.out.print("Asientos disponibles: ");
        miBus.mostrarAsientosDisponibles();

        System.out.println("\n--- Suben 45 pasajeros (solo caben 40) ---");
        miBus.subir(45);
        miBus.cobrarPasaje(45);
        System.out.println("Pasajeros actuales: " + miBus.getPasajerosActuales());
        System.out.printf("Recaudacion total: $%.2f%n", miBus.getRecaudacion());
        System.out.print("Asientos disponibles: ");
        miBus.mostrarAsientosDisponibles();
    }
}

File: 1) INTRODUCCION_A_POO/Ejercicio-2/PYTHON/Bus.py
class Bus:
    def __init__(self, capacidad_maxima):
        self.capacidad_maxima = capacidad_maxima
        self.pasajeros_actuales = 0
        self.recaudacion = 0.0

    def subir(self, cantidad):
        if self.pasajeros_actuales + cantidad <= self.capacidad_maxima:
            self.pasajeros_actuales += cantidad
        else:
            self.pasajeros_actuales = self.capacidad_maxima

    def cobrar_pasaje(self, cantidad):
        self.recaudacion += cantidad * 1.50

    def mostrar_asientos_disponibles(self):
        print(self.capacidad_maxima - self.pasajeros_actuales)

mi_bus = Bus(50)

print(f"Capacidad Maxima del bus: {mi_bus.capacidad_maxima}")
print(f"Pasajeros iniciales: {mi_bus.pasajeros_actuales}")
print(f"Recaudacion inicial: ${mi_bus.recaudacion:.2f}")

print("\n--- Suben 10 pasajeros ---")
mi_bus.subir(10)
mi_bus.cobrar_pasaje(10)
print(f"Pasajeros actuales: {mi_bus.pasajeros_actuales}")
print(f"Recaudacion total: ${mi_bus.recaudacion:.2f}")
print("Asientos disponibles:")
mi_bus.mostrar_asientos_disponibles()
